src/geotiffwriter.js

```diff
@@ -251,17 +252,47 @@ const encodeImage = (values, width, height, metadata) => {
   }
 
   const prfx = new Uint8Array(encodeIfds([ifd]));
+  const samplesPerPixel = ifd[277];
 
-  const img = new Uint8Array(values);
+  const dataType = values.constructor.name;
+  const TypedArray = typeMap[dataType];
 
-  const samplesPerPixel = ifd[277];
+  let elementSize = 4;
+  if (TypedArray) {
+    elementSize = TypedArray.BYTES_PER_ELEMENT;
+  }
+
+  const data = new Uint8Array(numBytesInIfd + (values.length * elementSize * samplesPerPixel));
 
-  const data = new Uint8Array(numBytesInIfd + (width * height * samplesPerPixel));
   times(prfx.length, (i) => {
     data[i] = prfx[i];
   });
-  forEach(img, (value, i) => {
-    data[numBytesInIfd + i] = value;
+
+  forEach(values, (value, i) => {
+    if (!TypedArray) {
+      data[numBytesInIfd + i] = value;
```

Comment:
I think this would only work if the Array of values is within the 0-255 range.  Is there any way we could update this to support Float64 by default?  Could we do `view.setFloat64` to support any floating point number?

Open to your thoughts!